### PR TITLE
mergiraf: init at 0.2.0

### PR DESCRIPTION
**Pull request by zimbatm**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/me/mergiraf/package.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# native check inputs
+, git
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "mergiraf";
+  version = "0.2.0";
+
+  src = fetchzip {
+    urls = [ "https://codeberg.org/mergiraf/mergiraf/archive/v${version}.tar.gz" ];
```
[the remark this answers is not in the document]
Comment (zimbatm) [reply pkgs/by-name/me/mergiraf/package.nix]: good question. fixed

**Review by kwaa — APPROVED** [review]

File: pkgs/by-name/me/mergiraf/package.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "tree-sitter-go-0.23.1" = "sha256-elPqkvVYs0vADOuN/umDteWP5hqcXhQAoSkqYDtTxaU=";
+      "tree-sitter-xml-0.7.0" = "sha256-RTWvOUAs3Uql9DKsP1jf9FZZHaZORE40GXd+6g6RQZw=";
+      "tree-sitter-yaml-0.6.1" = "sha256-gS+SjOnGl/86U9VV/y1ca7naYIe7DAOvOv++jCRLTKo=";
+    };
+  };
+
+  nativeCheckInputs = [ git ];
+
+  meta = {
+    description = "A syntax-aware git merge driver for a growing collection of programming languages and file formats";
```
Comment (Aleksanaa) [inline pkgs/by-name/me/mergiraf/package.nix:35]:
```suggestion
    description = "Syntax-aware git merge driver for a growing collection of programming languages and file formats";
```

https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

(new rule added half an year ago)

File: pkgs/by-name/me/mergiraf/package.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    mainProgram = "mergiraf";
+    homepage = "https://mergiraf.org/";
+    license = lib.licenses.gpl3;
```
Comment (Aleksanaa) [inline pkgs/by-name/me/mergiraf/package.nix:38]:
```suggestion
    license = lib.licenses.gpl3Only;
```

https://codeberg.org/mergiraf/mergiraf/src/branch/main/Cargo.toml#L5

(`licenses.gpl3` is deprecated)

**Review by Aleksanaa — APPROVED** [review]

**Comment by zimbatm** [comment]
Thanks, everyone, for your reviews. I will assume that everything has been fixed now and merge this.

**Pull request merged**